File: android/sandwich/src/main/kotlin/io/qonversion/sandwich/NoCodesEventListener.kt
package io.qonversion.sandwich

interface NoCodesEventListener {
    fun onNoCodesEvent(event: Event, payload: BridgeData? = null)
    
    enum class Event(val key: String) {
        ScreenShown("nocodes_screen_shown"),
        Finished("nocodes_finished"),
        ActionStarted("nocodes_action_started"),
        ActionFinished("nocodes_action_finished"),
        ActionFailed("nocodes_action_failed"),
        ScreenFailedToLoad("nocodes_screen_failed_to_load")
    }
} 

File: ios/sandwich/NoCodesEventListener.swift
//
//  NoCodesEventListener.swift
//  QonversionSandwich
//
//  Created by Suren Sarkisyan on 08.05.2025.
//  Copyright © 2025 Qonversion Inc. All rights reserved.
//

import Foundation

@objc public protocol NoCodesEventListener {
    @objc func noCodesDidTrigger(event: String, payload: [String: Any]?)
}

public enum NoCodesEvent: String {
    case screenShown = "nocodes_screen_shown"
    case finished = "nocodes_finished"
    case actionStarted = "nocodes_action_started"
    case actionFailed = "nocodes_action_failed"
    case actionFinished = "nocodes_action_finished"
    case screenFailedToLoad = "nocodes_screen_failed_to_load"
}
